### PR TITLE
Fix access to redirected git repositories

### PR DESCRIPTION
**Pull request by ts-schu**
spring-cloud-config-server configured to use a git repository throws `TransportException with error code 302 Found` when that repository gets redirected (e.g. moved to a diferent group/organization).

Fixes [issue 2123](https://github.com/spring-cloud/spring-cloud-config/issues/2123)

Version : v3.1.3
Module : Spring-Cloud-Config-Server

### CODE REVIEW

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ private HttpClientBuilder lookupHttpClientBuilder(final URL url) {
 
 		if (builderMap.isEmpty()) {
 			this.log.warn(String.format("No custom http config found for URL: %s", url));
-			return HttpClients.custom();
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java:102]: Why this change?

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/support/HttpClientSupport.java [hunk 2]
```diff
@@ -65,6 +65,8 @@ public static HttpClientBuilder builder(HttpEnvironmentRepositoryProperties envi
 			httpClientBuilder.setDefaultCredentialsProvider(new SystemDefaultCredentialsProvider());
 		}
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/support/HttpClientSupport.java:67]: A comment here indicating JGit is handling redirects would be good

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java (hunk 1 above)
Comment (ts-schu) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java:102]: I changed the code to give jGit the oppertinity to use their own HttpClient for standard cases. It is not necessary but prevents incompatible configuration of HttpClient between jGit and SpringCloudConfig

Comment (ts-schu) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java:102]: I can drop that change (and the related changes to the unit test).

Comment (ts-schu) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java:102]: Good idea, but

- using HttpClientSupport needs either JGitEnvironmentProperties or MultipleJGitEnvironmentProperties and
- has to be handling GeneralSecurityException

IMO the effort is too much in comparison to letting jGit handle the creation of the HttpClient

Comment (ryanjbaxter) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java:102]: OK please ad a comment that we are letting JGit handle it
